toniblyx/prowler
# fix(aws): Handle TZ while refreshing boto3 credentials

### Context

Right now Boto3 returns the credentials expiration when assuming an IAM Role with UTC as the timezone but Prowler works with the local timezone configured in the workstation.

### Description

Workaround to fix the issue checking `if refreshed_credentials["expiry_time"] <= datetime.now(get_localzone())`. Needs further testing and unit tests for the `refresh_credentials` function in the `AWSProvider` class.

### How to test it
If the following command is launched on master, you will see that each AWS API call forces the credentials refresh
`./prowler.py --profile <PROFILE> --role arn:aws:iam::<ACCOUNT>:role/ProwlerRole --external-id <EXTERNAL_ID> --log-level INFO --region eu-west-1 --session-duration 900`. Fixed in this branch.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
